codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/EnumTraitValidationErrorMessage.kt

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.server.smithy
+
+import software.amazon.smithy.model.shapes.Shape
+import software.amazon.smithy.model.traits.EnumTrait
+import software.amazon.smithy.rust.codegen.server.smithy.generators.idWithHashEscaped
+
+fun EnumTrait.validationErrorMessage() =
+    "Value at '{}' failed to satisfy constraint: Member must satisfy enum value set: [${enumValueSet()}]"
+
+fun EnumTrait.shapeConstraintViolationDisplayMessage(shape: Shape) =
+    "Value provided for '${shape.idWithHashEscaped()}' failed to satisfy constraint: Member must satisfy enum value set: [${enumValueSet()}]"
```

Comment:
I'd argue that it's not this function's responsibility to escape the hash, but rather the caller's, who is using `RustWriter`.

Reply:
Yes, you have a valid point. Changing it to not escape it here.